model.rb

```diff
@@ -38,40 +53,114 @@ def create_managers_table
+	# Adds a new manager to database
 	def add_manager(name, pin)
 		db.execute("INSERT INTO managers (name,pin) VALUES (?,?)", name, pin)
 	end
 
-	# Adds new account with given name and pin, sets balance to 0
-	def add_account(name, pin)
-		db.execute("INSERT INTO accounts (name,balance,pin) VALUES (?,?,?)", name, 0, pin)
-	end
-
-	# Returns true if name and pin match are found or false if not found; table parameter should be either "accounts" or "managers"
+	# Returns true if name and pin match are found or false if not found; table 
+	# parameter should be either "customers" or "managers"
 	def verify_pin(name, pin, table)
-		db.execute("SELECT 1 FROM #{table} WHERE name = ? AND pin = ?", name, pin).length > 0
+		db.execute("SELECT 1 FROM #{table} WHERE name = ? AND pin = ?", 
+			name, pin).length > 0
 	end
 
-	# Returns list of all bank accounts as an array of hashes. Requires manager name and pin match to be true
+	# Returns list of all bank accounts as an array of hashes. Requires manager 
+	# name and pin match to be true
 	def account_list(name, pin)
 		list = db.execute("SELECT * FROM accounts")
 		return list if verify_pin(name, pin, "managers")
 	end
 
-	# Returns balance for an account
-	def balance(name, pin)
-		customer = db.execute("SELECT * FROM accounts WHERE name = ? AND pin = ?", name, pin)
-		return customer[0]["balance"]
+	# Returns true if name exists in given table; table parameter should be either
+	# "customers" or "managers"
+	def name_exists?(name, table)
+		db.execute("SELECT 1 FROM #{table} WHERE name = ?", 
+			name).length > 0
 	end
 
-# Subtracts amount from balance of an account
-	def withdraw(name, pin, amount)
-		db.execute("UPDATE accounts SET balance = balance - ? WHERE name = ? AND pin = ?", amount, name, pin)
+	# Returns customer_id of customer found with name and pin
+	def find_customer_id(name, pin)
+		customer = db.execute("SELECT * FROM customers WHERE name = ? AND pin = ?", 
+			name, pin)
+		return customer[0]["customer_id"]
 	end
 
-# Add amount to balance of an account
-	def deposit(name, pin, amount)
-		db.execute("UPDATE accounts SET balance = balance + ? WHERE name = ? AND pin = ?", amount, name, pin)
+	# Returns accounts for customer
+	def customer_accounts(name, pin)
+		db.execute("SELECT * FROM accounts WHERE customer_id = ?", find_customer_id(
+			name, pin))
+	end
+
+	# Returns customer found with given customer_id
+	def find_customer_by_customer_id(customer_id)
+		db.execute("SELECT * FROM customers WHERE customer_id = ?",
+			customer_id)
+	end
+
+	# Returns Customer instance with customer from database
+	def load_customer(customer_id)
+		customer = find_customer_by_customer_id(customer_id)
+		Customer.new(db, customer[0]["name"], customer[0]["pin"])
+	end
+
+	# Returns Account instance with account from database
+	def load_account(account_id)
+		account = db.execute("SELECT * FROM accounts WHERE account_id = ?",
+			account_id)
+		Account.new(db, account[0]["customer_id"], false)
```

Comment:
My bad, replace this line with the following code. This will fix balance resetting issue.

```
new_instance = Account.new(db, account[0]["customer_id"], false)
new_instance.balance = account[0]["balance"]
new_instance
```